projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java

```diff
@@ -135,6 +138,13 @@ public boolean buildImage(
       }
     }
 
+    // TODO: Remove when deploymentDataJobBaseImage deprecated.
+    if (jobDeployment.getPythonVersion() == null && deploymentDataJobBaseImage == null) {
```

Comment:
why would a jobDeployment have a null python version?

Reply:
In cases when no python_version is passed to the request, or edge cases where either the supportedPythonVersions or defaultPythonVersion are not properly set. We need to add configuration validation for the properties, but this would be done in a separate PR.